include/boost/random/niederreiter_base2.hpp

```diff
@@ -0,0 +1,391 @@
+/* boost random/nierderreiter_base2.hpp header file
+ *
+ * Copyright Justinas Vygintas Daugmaudis 2010-2017
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE_1_0.txt or copy at
+ * http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_RANDOM_NIEDERREITER_BASE2_HPP
+#define BOOST_RANDOM_NIEDERREITER_BASE2_HPP
+
+#include <boost/random/detail/gray_coded_qrng_base.hpp>
+#include <boost/random/detail/config.hpp>
+#include <boost/random/detail/operators.hpp>
+
+#include <limits>
+#include <boost/cstdint.hpp>
+
+#include <boost/dynamic_bitset.hpp>
+#include <boost/multi_array.hpp>
+
+//!\file
+//!Describes the quasi-random number generator class template niederreiter_base2.
+//!
+//!\b Note: it is especially useful in conjunction with class template uniform_real.
+
+namespace boost {
+namespace random {
+
+/** @cond */
+namespace detail {
+namespace nb2 {
+
+/*
+  Primitive polynomials in binary encoding
+  {
+    { 1, 0, 0, 0, 0, 0 },    1
+    { 0, 1, 0, 0, 0, 0 },    x
+    { 1, 1, 0, 0, 0, 0 },    1 + x
+    { 1, 1, 1, 0, 0, 0 },    1 + x + x^2
+    { 1, 1, 0, 1, 0, 0 },    1 + x + x^3
+    { 1, 0, 1, 1, 0, 0 },    1 + x^2 + x^3
+    { 1, 1, 0, 0, 1, 0 },    1 + x + x^4
+    { 1, 0, 0, 1, 1, 0 },    1 + x^3 + x^4
+    { 1, 1, 1, 1, 1, 0 },    1 + x + x^2 + x^3 + x^4
+    { 1, 0, 1, 0, 0, 1 },    1 + x^2 + x^5
+    { 1, 0, 0, 1, 0, 1 },    1 + x^3 + x^5
+    { 1, 1, 1, 1, 0, 1 },    1 + x + x^2 + x^3 + x^5
+    { 1, 1, 1, 0, 1, 1 }     1 + x + x^2 + x^4 + x^5
+  };
+*/
+
+// Maximum allowed space dimension
+#define BOOST_RANDOM_NIEDERREITER_BASE2_MAX_DIMENSION 54
```

Comment:
Can the number of dimensions supported be increased? I personally am looking towards 30,000 dimensions.

Reply:
This is again better than the GSL, which is only implemented up to dimension 12.